dask/dataframe/core.py

```diff
@@ -4555,14 +4556,80 @@ def sort_values(
             **kwargs,
         )
 
+    def _check_index(self, other: Any) -> Optional[Any]:
+        """Internal method to check the column passed to set_index.
+
+        Returns None if this column is already the index.
+        """
+
+        # Check other can be translated to column name or column object
+        index = other
```

Comment:
Why rename this here? It looks like `other` is never used in this function now?

Reply:
Could be done with `other`as well probably. I'll check!

Reply:
I've finally left this as `sanitized_other`, because the checking function sometimes flattens the list passed.

Reply:
As I said in the top-level comment, I'm not sure it's a good idea for a function named `_check_index()` to return a modified copy of the index (or return anything, really)...

Reply:
I've refactored the code so as not to have an auxiliary method and thus `return self` directly. Please take a look.